### ISSUE TITLE
stability issue from small differences in initialization

### ISSUE THREAD

**Issue by Nathan-Becker** [glorialulu/wmoc#9]
I'm having some trouble getting stable results out of the simulations. I'm attaching an example code below. It reads from two versions of the wntr_network input file. Both versions are identical except for a slightly different pump curve. For pump_curve_1 I see crazy pressure fluctuations after the valve is opened. But for pump_curve_2 I see smooth pressure oscillations that are much more physically plausible. Shown below:

![pressure](https://user-images.githubusercontent.com/53197475/91502350-b34d5580-e87c-11ea-9a89-8b40920a8102.png)

If I change the valve_open_period (at the beginning of the program) from 0.5 to 0.6, then the behavior switches and pump_curve_1 produces more stable results than pump_curve_2:
![pressure_b](https://user-images.githubusercontent.com/53197475/91502595-4edec600-e87d-11ea-9a2a-5e4a601d4701.png)

How can I consistently achieve stable results?

Here is the test program:
```
valve_open_period=0.5


import matplotlib
matplotlib.use('agg')
from matplotlib import pyplot as plt
import numpy
import wntr
import tsnet


def m3s_to_gpm(m3s):
    return m3s*(60*264.172)

def m_to_psi(m):
    return m*1.4219702063247


# Restore model for each pump
wn1 = wntr.network.WaterNetworkModel('pumpcurve_1.inp')
wn2 = wntr.network.WaterNetworkModel('pumpcurve_2.inp')

# Access pump curve 1
pts_1 = wn1.get_link('supply').get_pump_curve().points
coeffs_1 = wn1.get_link('supply').get_head_curve_coefficients()
q_1 = numpy.linspace(pts_1[0][0],pts_1[-1][0])
H_1 = coeffs_1[0] - coeffs_1[1]*(q_1**coeffs_1[2])

# Access pump curve 2 
pts_2 = wn2.get_link('supply').get_pump_curve().points
coeffs_2 = wn2.get_link('supply').get_head_curve_coefficients()
q_2 = numpy.linspace(pts_2[0][0],pts_2[-1][0])
H_2 = coeffs_2[0] - coeffs_2[1]*(q_2**coeffs_2[2])

# Plot pump curves
plt.plot(m3s_to_gpm(q_1), m_to_psi(H_1), label='pumpcurve_1')
plt.plot(m3s_to_gpm(q_2), m_to_psi(H_2), label='pumpcurve_2')
plt.xlabel('gpm')
plt.ylabel('PSI')
plt.draw()
plt.legend(loc='best')
plt.savefig('pumpcurves.png')

# Define valve curve
valve_curve = [(100.0, 0.0008099981681764881), (90.0, 0.00040499908408824403), (80.0, 0.00020249954204412202), (70.0, 0.00010124977102206101), (60.0, 5.394587800055411e-05), (50.0, 2.7539937718000598e-05), (40.0, 1.6199963363529762e-05), (30.0, 9.007179630122548e-06), (20.0, 5.0705885327848155e-06), (10.0, 2.7053938817094702e-06), (0.0, 0.0)]

# Simulate using each pump curve
time_index = []
pressure = []
flowrate = []
for input_name in ['pumpcurve_1', 'pumpcurve_2']:
    # Construct transient model
    tm = tsnet.network.TransientModel(input_name+'.inp')

    # Set wavespeed
    tm.set_wavespeed(1200) # m/s

    # Set time options
    tf = 4   # simulation period [s]
    dt = tsnet.network.discretize.max_time_step(tm)
    tm.set_time(tf, dt)

    # Set valve opening
    tc = valve_open_period # valve opening period [s]
    ts = 1 # valve opening start time [s]
    se = 1 # end open fraction (ranges from 0 to 1)
    m = 1 # open constant [dimensionless]
    tm.valve_opening('valve_3',[tc,ts,se,m], curve=valve_curve)

    # Initialize steady state simulation
    tm = tsnet.simulation.Initializer(tm, 0, 'DD')

    # Transient simulation
    tm = tsnet.simulation.MOCSimulator(tm, 'temp.res', 'unsteady')

    # Save results
    time_index.append(numpy.asarray(tm.simulation_timestamps, dtype=float))
    pressure.append(m_to_psi(numpy.asarray(tm.get_node('ref').head, dtype=float)))
    flowrate.append(m3s_to_gpm(numpy.asarray(tm.get_link('to_ref').start_node_flowrate, dtype=float)))

# Prep plotting
fig = plt.figure(figsize=(8,5), dpi=80, facecolor='w', edgecolor='k')

# Plot pressure
plt.clf()
plt.plot(time_index[0], pressure[0], label='pump_curve_1', linewidth=1.5)
plt.plot(time_index[1], pressure[1], label='pump_curve_2', linewidth=1.5)
plt.ylabel("Pressure [PSI]")
plt.xlim([0,4])
plt.ylim([0,100])
plt.xlabel("Time [s]")
plt.legend(loc='best')
plt.draw()
plt.savefig('pressure.png')


# Plot flow-rates
plt.clf()
plt.plot(time_index[0], flowrate[0], label='pump_curve_1', linewidth=1.5)
plt.plot(time_index[1], flowrate[1], label='pump_curve_2', linewidth=1.5)
plt.ylabel("flow-rate [gpm]")
plt.xlim([0,4])
plt.ylim([0,3])
plt.xlabel("Time [s]")
plt.legend(loc='best')
plt.draw()
plt.savefig('flowrate.png')
```

Here are the wntr input files:

[pumpcurve_1.txt](https://github.com/glorialulu/TSNet/files/5139067/pumpcurve_1.txt)
[pumpcurve_2.txt](https://github.com/glorialulu/TSNet/files/5139069/pumpcurve_2.txt)

**Comment by Nathan-Becker** [comment]
Any suggestions for mitigating this issue?

**Comment by glorialulu** [comment]
Hi Nathan, 

The stability issue is with the numerical scheme of the unsteady friction model, which is known and we reported that in the paper. We will try to resolve the issue in future releases. 

Meanwhile, one way to avoid this issue to use the steady friction model. `tm = tsnet.simulation.MOCSimulator(tm, 'temp.res', 'steady')`. This model is stable but it can underestimate the damping effects; however, the difference is relatively small in your case. Also, if your project focuses on the amplitude of the first wave of change, the steady model should be good enough and much faster than the unsteady model.  I attached the results I got using the steady friction model. 
![image](https://user-images.githubusercontent.com/32466099/94188096-833ea580-fe6e-11ea-9f70-4428c1e55539.png)
